include/pros/rtos.hpp

```diff
@@ -78,6 +79,53 @@ class Task {
 	 */
 	Task(task_fn_t function, void* parameters = NULL, const char* name = "");
 
+	/**
+	 * Creates a new task and add it to the list of tasks that are ready to run.
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENOMEM - The stack cannot be used as the TCB was not created.
+	 *
+	 * \param function
+	 *        Callable object to use as entry function
+	 * \param prio
+	 *        The priority at which the task should run.
+	 *        TASK_PRIO_DEFAULT plus/minus 1 or 2 is typically used.
+	 * \param stack_depth
+	 *        The number of words (i.e. 4 * stack_depth) available on the task's
+	 *        stack. TASK_STACK_DEPTH_DEFAULT is typically sufficienct.
+	 * \param name
+	 *        A descriptive name for the task.  This is mainly used to facilitate
+	 *        debugging. The name may be up to 32 characters long.
+	 *
+	 */
+	template <class F>
+	Task(F&& function, std::uint32_t prio = TASK_PRIORITY_DEFAULT,
+	     std::uint16_t stack_depth = TASK_STACK_DEPTH_DEFAULT, const char* name = "")
+	     : Task(&[] (void* parameters) {
+		auto ptr = std::static_cast<std::function<void()>*>(parameters);
+		(*ptr)();
+		delete ptr;
+	     }, new std::function<void()>(std::forward<F>(function))) {}
```

Comment:
you can probably specify a parameter pack above in the constructor and pass them on down here
e.g.
```cpp
template<class F, class ... Args>
Task(F&& function, Args... args) : Task(<lambda>, <forward_func>, &args...) {};
```
so that we can avoid introducingg [more] ambiguous constructors and also only have to write one constructor for this.